### DISCUSSION TITLE
How to make the window non-resizable while automatic layout is enabled?

### DISCUSSION THREAD

**Discussion by peterjc123** [slint-ui/slint#2514] [Q&A] [answered]
Alternatively, can I get the current width and height of the widgets under automatic layout?

**Comment by ogoffart** [comment, marked as answer]
You can use the `width` and `height` properties of the window to make it fix, and the `preferred-width` / `preferred-height` properties of a layout to know the size it would have.

So I think you want something like this:

```
export component Demo {
    height: layout.preferred-height;
    width: layout.preferred-width;
    layout := VerticalBox {
        Text { text: "Hello World!"; }
        TextEdit { }
        Button { text: "Ok"; }
    }
}
```

[demo](https://slint-ui.com/editor/?snippet=import+%7B+Button%2C+VerticalBox+%2C+TextEdit%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++height%3A+layout.preferred-height%3B%0A++++width%3A+layout.preferred-width%3B%0A++++layout+%3A%3D+VerticalBox+%7B%0A++++++++Text+%7B+text%3A+%22Hello+World%21%22%3B+%7D%0A++++++++TextEdit+%7B+%7D%0A++++++++Button+%7B+text%3A+%22Ok%22%3B+%7D%0A++++%7D%0A%7D%0A%0A)
